### ISSUE TITLE
Composer run-script flawed <args> handling

### ISSUE THREAD

**Issue by ktomk** [composer/composer#6225]
My `composer.json`:

```json
...
  "scripts": {
    "test": "phpunit"
  }
}
```

Output of `composer diagnose`:

```
Checking composer.json: OK
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com oauth access: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
```

When I run this command:

```
$ composer run-script -- test --help
```

I get the following output:

```
Usage:
  run-script [options] [--] [<script>] [<args>]...

Arguments:
  script                         Script name to run.
  args                           

Options:
      --timeout=TIMEOUT          Sets script timeout in seconds, or 0 for never.
      --dev                      Sets the dev mode.
      --no-dev                   Disables the dev mode.
  -l, --list                     List scripts.
  -h, --help                     Display this help message
  -q, --quiet                    Do not output any message
  -V, --version                  Display this application version
      --ansi                     Force ANSI output
      --no-ansi                  Disable ANSI output
  -n, --no-interaction           Do not ask any interactive question
      --profile                  Display timing and memory usage information
      --no-plugins               Whether to disable plugins.
  -d, --working-dir=WORKING-DIR  If specified, use the given directory as working directory.
  -v|vv|vvv, --verbose           Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug

Help:
  The run-script command runs scripts defined in composer.json:
  
  php composer.phar run-script post-update-cmd

```

And I expected this to happen:

Help information from phpunit

I expect this because that's how it normally works (by convention), then how it should work (by the composer help message) and also because it's written in the composer docs.

**Comment by ktomk** [comment]
I also just tried the scenario in composer/composer master after updating my local repo, it's the same (the test script is phpunit in composer master).

Looks like the `--help` argument is consumed by composer despite `--` should have stopped it.

**Comment by ktomk** [comment]
The flaw is in the underlying Symfony Console component, it is very eager on `--help` and `-h` and if found in any entry of the `$argv` array it will replace the `run-script` command with Symfony console `help` command.

**Comment by ktomk** [comment]
I couldn't find an existing issue upstream, so I created one: https://github.com/symfony/symfony/issues/21869

**Comment by Seldaek** [comment]
As you've seen in symfony repo, this was fixed already, we just can't get the fix until we drop 5.3/5.4.
